src/handlers/comment/handlers/review.ts

```diff
@@ -0,0 +1,196 @@
+import { getBotContext, getLogger } from "../../../bindings";
+import { GPTResponse, Payload, StreamlinedComment, UserType } from "../../../types";
+import { getAllIssueComments, getAllLinkedIssuesAndPullsInBody, getPullByNumber } from "../../../helpers";
+import { CreateChatCompletionRequestMessage } from "openai/resources/chat";
+import { askGPT, decideContextGPT, gptContextTemplate, specCheckTemplate, speckCheckResponse } from "../../../helpers/gpt";
+
+/**
+ * @notice Three calls to Chad are made. First for context, second for review and third for finalization.
+ * @returns Pull Request Report
+ */
+export const review = async (body: string) => {
+  const context = getBotContext();
+  const logger = getLogger();
+
+  const payload = context.payload as Payload;
+  const issue = payload.issue;
+
+  if (!issue) {
+    return `Payload issue is undefined`;
+  }
+
+  if (!body) {
+    return `Payload body is undefined`;
+  }
+
+  const isPr = await getPullByNumber(context, issue.number);
+
+  if (!isPr) {
+    return `This command can only be used on pull requests.`;
+  }
+
+  const reviewRegex = /^\/review/;
+  const reviewRegexMatch = body.match(reviewRegex);
+
+  if (!reviewRegexMatch) {
+    return `Payload body does not match regex`;
+  }
+
+  const streamlined: StreamlinedComment[] = [];
+  let chatHistory: CreateChatCompletionRequestMessage[] = [];
+  let linkedPRStreamlined: StreamlinedComment[] = [];
+  let linkedIssueStreamlined: StreamlinedComment[] = [];
+
+  const comments = await getAllIssueComments(issue.number);
+  const commentsRaw = await getAllIssueComments(issue.number, "raw");
+
+  if (!commentsRaw) {
+    logger.info(`Error getting issue comments`);
+    return `Error getting issue comments`;
+  }
+  streamlined.push({
+    login: issue.user.login,
+    body: issue.body,
+  });
+
+  comments.forEach(async (comment, i) => {
+    if (comment.user.type == UserType.User || commentsRaw[i].body.includes("<!--- { 'UbiquityAI': 'answer' } --->")) {
+      streamlined.push({
+        login: comment.user.login,
+        body: comment.body,
+      });
+    }
+  });
+
+  // returns the conversational context from all linked issues and prs
+  const links = await getAllLinkedIssuesAndPullsInBody(issue.number);
+
+  if (typeof links === "string") {
+    logger.info(`Error getting linked issues or prs: ${links}`);
+    return `Error getting linked issues or prs: ${links}`;
```

Comment:
Cool to see the error. I believe we are using an error render function now for comments to display as errors.